ui/src/components/AttendAddEdit.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useEffect, useState } from 'react';
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FinalFacDashProp } from './FDashboard';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import { useUserdetails } from './UserContext';
+
+
+interface AddEditProps {
+    initialData ?: FinalFacDashProp;
+    handle : ()=> void;
+ }
+
+type subType = {
+  id : number,
+  code : string,
+  name : string,
+}
+
+type subTypes = subType[];
+
+const AttendAddEdit = ({initialData, handle} : AddEditProps) => {
+const [subdetails,SetSubdetails] = useState<subTypes>([])
+  const {userdetails} = useUserdetails();
+  useEffect(
+    () => {
+        (async () =>{
+         await axios.get(`/subject/all`)
+            .then((response) => {
+              if (response.status === 200){
+                SetSubdetails(response.data);
+              };
+             
+            })
+            .catch((error) => {
+              toast.error("Subject loading failed")
```

Comment:
API calls should be inside respective service for each model. Move all API calls to services and call them from there